app/code/Magento/Cms/Block/BlockByIdentifier.php

```diff
@@ -0,0 +1,170 @@
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Cms\Block;
+
+use Magento\Cms\Api\Data\BlockInterface;
+use Magento\Cms\Api\GetBlockByIdentifierInterface;
+use Magento\Cms\Model\Block as BlockModel;
+use Magento\Cms\Model\Template\FilterProvider;
+use Magento\Framework\DataObject\IdentityInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\AbstractBlock;
+use Magento\Framework\View\Element\Context;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * This class is replacement of \Magento\Cms\Block\Block, that accepts only `string` identifier of CMS Block
+ */
+class BlockByIdentifier extends AbstractBlock implements IdentityInterface
+{
+    public const CACHE_KEY_PREFIX = 'CMS_BLOCK';
+
+    /**
+     * @var GetBlockByIdentifierInterface
+     */
+    private $blockByIdentifier;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var FilterProvider
+     */
+    private $filterProvider;
+
+    /**
+     * @var BlockInterface
+     */
+    private $cmsBlock;
+
+    /**
+     * @param GetBlockByIdentifierInterface $blockByIdentifier
+     * @param StoreManagerInterface $storeManager
+     * @param FilterProvider $filterProvider
+     * @param Context $context
+     * @param array $data
+     */
+    public function __construct(
+        GetBlockByIdentifierInterface $blockByIdentifier,
+        StoreManagerInterface $storeManager,
+        FilterProvider $filterProvider,
+        Context $context,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->blockByIdentifier = $blockByIdentifier;
+        $this->storeManager = $storeManager;
+        $this->filterProvider = $filterProvider;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function _toHtml(): string
+    {
+        try {
+            return $this->filterOutput(
+                $this->getCmsBlock()->getContent()
+            );
+        } catch (NoSuchEntityException $e) {
+            return '';
+        }
+    }
+
+    /**
+     * Returns the value of `identifier` injected in `<block>` definition
+     *
+     * @return string|null
+     */
+    private function getIdentifier(): ?string
+    {
+        return $this->getdata('identifier') ?: null;
+    }
+
+    /**
+     * Filters the Content
+     *
+     * @param string $content
+     * @return string
+     * @throws NoSuchEntityException
+     */
+    private function filterOutput(string $content): string
+    {
+        return $this->filterProvider->getBlockFilter()
+            ->setStoreId($this->getCurrentStoreId())
+            ->filter($content);
+    }
+
+    /**
+     * Loads the CMS block by `identifier` provided as an argument
+     *
+     * @return BlockInterface|BlockModel
+     * @throws NoSuchEntityException
+     */
+    private function getCmsBlock(): BlockInterface
```

Comment:
`private` prevents extensibility in a big way, rather use `protected` so that (for whatever reason) this class can be extended upon at least through a preference if really necessary. It's unfortunate that plugins can't be used on protected methods.

Reply:
Using protected methods is discouraged and is going to fail Static code checks / Magento health check.

Also please consider reviewing these resources:
- https://devdocs.magento.com/guides/v2.3/ext-best-practices/extension-coding/common-programming-bp.html#composition-over-inheritance
- https://github.com/magento/magento2/issues/1019#issuecomment-72757989

So I am not going to change this into `protected` method as it's against what I consider as best practices.